packages/convex-helpers/server/customFunctions.ts

```diff
@@ -501,8 +511,8 @@ export type CustomBuilder<
   Visibility extends FunctionVisibility,
 > = {
   <
-    ArgsValidator extends PropertyValidators | void,
-    ReturnsValidator extends GenericValidator | void,
+    ArgsValidator extends PropertyValidators | GenericValidator | void,
+    ReturnsValidator extends PropertyValidators | GenericValidator | void,
```

Comment:
For ArgsValidator, I don't think we're handling generic validator correctly yet? we're just spreading in ...fn.args when now it could be a validator object (or even a union of objects?) where adding extra args will need some extra handling in those cases

Reply:
Ooh yeah you're totally right. I'll probably punt on supporting `GenericValidator` for now and follow up with a `mergeObjects` helper